File: grifballwebapp.client/src/app/api/dtos/signupResponseDto.ts
import { DateTime } from "luxon";

export class SignupResponseDto {
  seasonID: number = 0;
  userID: number = 0;
  personName: string | null = null;
  timeStamp!: DateTime;
  teamName: string | null = null;
  willCaptain: boolean = false;
  requiresAssistanceDrafting: boolean = false;
  timeslots: TimeslotDto[] = [];
}

export class TimeslotDto {
  id: number = 0;
  dayOfWeek: number = 0;
  time: string = '';
  isChecked: boolean = false;
  isDisabled: boolean = false;
  isHeader: boolean = false;
}

File: GrifballWebApp.Server/Signups/SignupResponseDto.cs
namespace GrifballWebApp.Server.Signups;

public class SignupResponseDto
{
    public int SeasonID { get; set; }
    public int UserID { get; set; }
    public string? PersonName { get; set; }
    public DateTime Timestamp { get; set; }
    public string? TeamName { get; set; }
    public bool WillCaptain { get; set; }
    public bool RequiresAssistanceDrafting { get; set; }
    public TimeslotDto[] Timeslots { get; set; } = [];
}

public class TimeslotDto
{
    public DayOfWeek DayOfWeek { get; set; }
    public TimeOnly Time { get; set; }
    public bool IsChecked { get; set; }
}
